### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by principlebrothers**
## Description
In this pull request my partner @alanluqman  and I @principlebrothers  refactored the `main.rb` and implemented 
- SOLID
- DRY
- YAGNI principles

### Requires
- Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
- Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: old_main.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# when 4
+
+# create_new_book
+
+# when 5
+
+# create_rental
+
+# when 6
+
```
Comment (phelian23) [inline old_main.md]: - [x] Kindly remove all commented code. Remember you can always use git commit history to check previous (old) code.

File: app.rb [hunk 2]
```diff
@@ -39,9 +39,9 @@ def list_of_people
...
       @people.each_with_index do |person, index|
         if person.instance_of?(Student)
-          puts "#{index + 1} -  [Student] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+          puts "#{index + 1} -  #{[person.class]} Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
         else
-          puts "#{index + 1} - [Teacher] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+          puts "#{index + 1} - #{[person.class]} Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
         end
       end
     end
```
Comment (phelian23) [inline app.rb:43]: - [x] Kindly refactor the this file by putting all requests for input and input validation in the startup.rb file that is dedicated for that purpose in your case. 
https://github.com/alanluqman/OOP-school-library/blob/1ead0a7cd235c075927134f9333a742cb73d3ed3/app.rb#L52-L60
https://github.com/alanluqman/OOP-school-library/blob/1ead0a7cd235c075927134f9333a742cb73d3ed3/app.rb#L66-L73
https://github.com/alanluqman/OOP-school-library/blob/1ead0a7cd235c075927134f9333a742cb73d3ed3/app.rb#L91-L103
https://github.com/alanluqman/OOP-school-library/blob/1ead0a7cd235c075927134f9333a742cb73d3ed3/app.rb#L105-L116
https://github.com/alanluqman/OOP-school-library/blob/1ead0a7cd235c075927134f9333a742cb73d3ed3/app.rb#L119-L123
https://github.com/alanluqman/OOP-school-library/blob/1ead0a7cd235c075927134f9333a742cb73d3ed3/app.rb#L135-L138
https://github.com/alanluqman/OOP-school-library/blob/1ead0a7cd235c075927134f9333a742cb73d3ed3/app.rb#L152-L153
https://github.com/alanluqman/OOP-school-library/blob/1ead0a7cd235c075927134f9333a742cb73d3ed3/app.rb#L152-L153

- [x] Also, the lines to create a new student/teacher should have been included in the create person method. 
https://github.com/alanluqman/OOP-school-library/blob/1ead0a7cd235c075927134f9333a742cb73d3ed3/app.rb#L78-L88

**Review by phelian23 — CHANGES_REQUESTED** [review]
Hi @Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
No linters errors ✔️
Descriptive pull request ✔️
Descriptive commit messages ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alanluqman** [comment]
Thanks, @phelian23, for the review to make us keep the code clean.  We have removed `old_main.md` file.
There are few things you request us to change that is contrary to the requirements from the previous project instruction. 
1. the methods you want us to move to the startup file is meant to be in the app.rb file.
2. we are told to review the main.rb file instead for today's work, not the app.rb

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @principlebrothers, @alanluqman 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.





 ## Highlights

- PR has a good title and summary :heavy_check_mark: 
- Correct Git flow :+1: 
- No linter error :+1: 

 ## Required Changes ♻️

- [x] Hi team, as requested by the previous reviewer, we need to make sure that we respect the Single responsibility rules of SOLID. Single-responsibility Principle (SRP) states that: **A class should have only one job** Right now your app class is requesting inputs from the user and also performing validations. I'll kindly advise you to create other classes/methods in another file for these aspects. Hence you must not move all these methods to the startup file, you can create another for this perspective. No input from the user or validations should be done in your app file. This way, your work will be cleaner and neater.

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi Team 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by principlebrothers** [comment]
@alanluqman  and @principlebrothers  appreciate the taken to review our work. Thank your @omarsalem7

**Pull request merged**